aten/src/THC/generic/THCTensorMathBlas.cu

```diff
@@ -433,6 +435,7 @@ static void THCTensor_(addmmImpl)(THCState *state, THCTensor *r_, THCTensor *t,
                    r__->stride((transpose_r == 'n' ? 1 : 0)));
 #endif
 
+#if !defined(THC_REAL_IS_BFLOAT16) || defined(__HIP_PLATFORM_HCC__)
```

Comment:
Why did these conditions need to change?

Reply:
If these conditions do not met, we do TORCH_CHECK(false, ...) in line 419 and the compiler complains the statements below are unreachable.

Reply:
That's a great point but TORCH_CHECK(false) is basically killing the process. Not sure what's PyTorch convention on whether we want to avoid leaks even when terminating the process.

Reply:
TORCH_CHECK is throwing a runtime error that can be caught and handled.

Reply:
Oh I see. I thought it's like assert. Thanks for letting me know and let me take a look to avoid leaks.